abci/abci.go

```diff
@@ -114,6 +135,15 @@ func (h *ProposalHandler) PrepareProposalHandler() sdk.PrepareProposalHandler {
 					sdkTxBytes[index] = txBz
 				}
 
+				if int64(gasLimit) > maxGasLimit {
```

Comment:
nit: Do we break early if the gasLimit == maxGasLimit?

Reply:
yea, we can break here.